[wip] Continue Refactoring

alexandercampbell opened the pull request:
FYI @pwittrock 

Feel free to scroll through this. Just mechanical changes and cleanup.
There are some design points I want to discuss Monday before I put serious effort into this repo.

pwittrock commented on cmd/apiserver-boot/boot/run.go:
```diff
@@ -63,20 +63,13 @@ func RunRun(cmd *cobra.Command, args []string) {
 		defer etcdCmd.Process.Kill()
 	}
 
-	time.Sleep(time.Second * 2)
```
These are actually needed right now, because each process needs to wait for the previous one to fully start.  This could be fixed with polling the health endpoints of the processes.

pwittrock commented:
one comment, otherwise looks good

alexandercampbell commented:
Issue with sleeps fixed.
Rebased onto master.
Fixed issue (58a1c82c3fca9f469cf75d89a4f98f41cd660697) that popped up due to recent double merge into master.

Should be good to go, PTAL
